### PR TITLE
Update environment variables passed to harv-install script

### PR DESCRIPTION
**Pull request by johnliu55tw**
Update the environment variables passed to the `harv-install` and `cos-installer-shutdown` script for `elemental` command. Do notice that elemental uses different names for several installation options, so we also need to adapt these new names in harv-install script.

~~We don't have to update the base-os version yet because `elemental` already exists in the current base-os: https://github.com/harvester/os2/releases/tag/20220407.~~
We do have to update the base OS version for the fix of https://github.com/rancher-sandbox/elemental/issues/164

# `HarvesterConfig.Install` fields and their corresponding environment variables
- ForceEFI: `ELEMENTAL_FORCE_EFI`
- Device: Positional arg or `ELEMENTAL_TARGET`
- ConfigURL: `ELEMENTAL_CLOUD_INIT`
- Silent: No env var. Must be specified by CLI arg `--quiet`
- ISOURL: `ELEMENTAL_ISO`
- PowerOff: `ELEMENTAL_POWEROFF`
- NoFormat: `ELEMENTAL_NO_FORMAT`
- Debug: No env var. Must be specified by CLI arg `--debug`
- TTY: `ELEMENTAL_TTY`
- ForceGPT: `ELEMENTAL_FORCE_GPT`

## Not in `HarvesterConfig.Install` but still required
- Custom partition layout file: `ELEMENTAL_PARTITION_LAYOUT`

# Todos
- [x] Test PXE installation
- [x] Air-gapped installation
- [x] Upgrade from older version of Harvester
- [x] Use `elemental` command to upgrade OS: https://github.com/harvester/harvester/pull/2166
    - Rebrand error: image
- [x] Fix `realpath: /run/netconfig/yp.conf: No such file or directory` 
    - `elemental` bug: https://github.com/rancher-sandbox/elemental/issues/164
    - Temporarily solved by add the symlink `/var/run -> /run` directly in `harv-install` script
    image
- [ ] (Will submit other PR) Move the environment variables adaptation code from `harv-install` to Go
- [ ] (Will submit other PR) Develop some automated tests for base OS verification

# Related issues
- https://github.com/harvester/harvester/issues/2133

# Dependencies
- https://github.com/harvester/os2/pull/36

### CODE REVIEW

File: package/harvester-os/files/usr/sbin/harv-install [hunk 1]
```diff
@@ -5,6 +5,13 @@ ISOMNT=/run/initramfs/live
 TARGET=/run/cos/target
 DATA_DISK_FSLABEL="HARV_LH_DEFAULT"
 
+# Update environment variables
```
Comment (johnliu55tw) [inline package/harvester-os/files/usr/sbin/harv-install:8]: `elemental` uses different environment variables compare to `HarvesterConfig.Install`, so we need to make some adaptations here. In the future, we would probably move these into `config.go`.

File: package/harvester-os/files/usr/sbin/harv-install [hunk 2]
```diff
@@ -352,27 +359,35 @@ do_data_disk_format()
     mkfs.ext4 -F -L "$DATA_DISK_FSLABEL" "$HARVESTER_DATA_DISK"
 }
 
+patch_root()
+{
+    # Patching some known issues exists in the rootfs (active.img)
+    # https://github.com/rancher-sandbox/elemental/issues/164
+    [ ! -e ${TARGET}/var/run ] && ln -s /run ${TARGET}/var/run
```
Comment (johnliu55tw) [inline package/harvester-os/files/usr/sbin/harv-install:366]: This patch could be removed once we have the latest `elemental` in hands. See https://github.com/rancher-sandbox/elemental/issues/164

**Comment by weihanglo** [comment]
Hi. Could you provide a link to changelog or elsewhere you found those changes?

**Comment by johnliu55tw** [comment]
> Hi. Could you provide a link to changelog or elsewhere you found those changes?

Names of those environment variables are automatically generated by [viper](https://github.com/spf13/viper) and [cobra](https://github.com/spf13/cobra/). For example [here](https://github.com/rancher-sandbox/elemental/blob/edc412b0dd1c219a07e023d16cafb6251f7280cd/cmd/install.go#L40) and [here](https://github.com/rancher-sandbox/elemental/blob/main/cmd/config/config.go). Currently, there is no document about this yet, so I manually tested all the environment variables.

**Review by bk201 — APPROVED** [review]
LGTM

**Review by guangbochen — APPROVED** [review]
LGTM, thanks for the update, in regards to the element environment variable description we can file an issue request of it, thanks.

**Pull request merged**